### PR TITLE
Adding the FormRecognizer Sdk

### PR DESCRIPTION
**Pull request by ramparab**

### CODE REVIEW

File: src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/Microsoft.Azure.CognitiveServices.Vision.FormRecognizer.csproj [hunk 1]
```diff
@@ -0,0 +1,29 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <!-- Please do not move/edit code below this line -->
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
+  <!-- Please do not move/edit code above this line -->
+
+  <PropertyGroup>
+    <PackageId>Microsoft.Azure.CognitiveServices.Vision.FormRecognizer</PackageId>
+    <Description>This client library provides access to the Microsoft Cognitive Services FormRecognizer APIs.</Description>
+    <Version>2.4.0-preview</Version>
```
Comment (dsgouda) [inline src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/Microsoft.Azure.CognitiveServices.Vision.FormRecognizer.csproj]: Versioning must start from `0.8.0-preview`

File: src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/Microsoft.Azure.CognitiveServices.Vision.FormRecognizer.csproj [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    <AssemblyName>Microsoft.Azure.CognitiveServices.Vision.FormRecognizer</AssemblyName>
+    <PackageTags>Microsoft Cognitive Services;Cognitive Services;Cognitive Services SDK;REST HTTP client;FormRecognizer;FormRecognizer API;FormRecognizer SDK;Vision;netcore451511</PackageTags>
+    <PackageReleaseNotes>
+    <![CDATA[
```
Comment (dsgouda) [inline src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/Microsoft.Azure.CognitiveServices.Vision.FormRecognizer.csproj]: Please move the description here

**Review by dsgouda — CHANGES_REQUESTED** [review]
Please create an `AssemblyInfo.cs` file similar to [this](https://github.com/Azure/azure-sdk-for-net/blob/88ca0663ab54391a60c59225080a1dbe0163de6c/src/SDKs/CognitiveServices/dataPlane/Vision/ContentModerator/ContentModerator/Properties/AssemblyInfo.cs) with proper versions

**Comment by dsgouda** [comment]
@ramparab Please fix the failing test cases too

File: src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/Microsoft.Azure.CognitiveServices.Vision.FormRecognizer.csproj (hunk 1 above)
Comment (ramparab) [reply src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/Microsoft.Azure.CognitiveServices.Vision.FormRecognizer.csproj]: done. Fixed to v1.0.0.0

File: src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/Microsoft.Azure.CognitiveServices.Vision.FormRecognizer.csproj (hunk 2 above)
Comment (ramparab) [reply src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/Microsoft.Azure.CognitiveServices.Vision.FormRecognizer.csproj]: done. Added assembly info file.

File: src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer.Tests/Microsoft.Azure.CognitiveServices.Vision.FormRecognizer.Tests.csproj [hunk 3]
```diff
@@ -0,0 +1,22 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+<Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.test.reference.props'))" />
+  <PropertyGroup>
+    <Description>Microsoft.Azure.CognitiveServices.Vision.FormRecognizer.Tests Class Library</Description>
+    <AssemblyName>Microsoft.Azure.CognitiveServices.Vision.FormRecognizer.Tests</AssemblyName>
+    <Version>1.0.0</Version>
```
Comment (dsgouda) [inline src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer.Tests/Microsoft.Azure.CognitiveServices.Vision.FormRecognizer.Tests.csproj]: We generally recommend starting versioning from `0.9.0-preview` for a brand new SDK

File: src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/generate.ps1 [hunk 4]
```diff
@@ -0,0 +1 @@
+Start-AutoRestCodeGeneration -ResourceProvider "cognitiveservices/data-plane/FormRecognizer" -AutoRestVersion "latest"
```
Comment (dsgouda) [inline src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/generate.ps1]: Was this code generated using the generate.ps1 script? If yes, there should be a .txt and SdkInfo_*.cs file generated, please commit these files too

**Review by dsgouda — CHANGES_REQUESTED** [review]

**Comment by dsgouda** [comment]
@ramparab Please link the REST spec PR too

File: src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/generate.ps1 (hunk 4 above)
Comment (ramparab) [reply src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/generate.ps1]: The code was not generated using generate.ps1. Once I am able to do that I will add those files you mention. The code is generated using AutoRest command separately.

**Comment by dsgouda** [comment]
Created for REST spec PR 
https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fpull%2F5684&data=02%7C01%7Cdeshank%40microsoft.com%7Caba79e4b137247f83eb208d6c75b62db%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636915592640488898&sdata=IJtzIQHtGMlq18WRV5TKIphQ3Pmav703q2G5KhOpbwA%3D&reserved=0

File: src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer.Tests/Microsoft.Azure.CognitiveServices.FormRecognizer.Tests.csproj [hunk 5]
```diff
@@ -0,0 +1,25 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+<Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.test.reference.props'))" />
```
Comment (dsgouda) [inline src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer.Tests/Microsoft.Azure.CognitiveServices.FormRecognizer.Tests.csproj]: For consistency please rename this file to `CognitiveServices.FormRecognizer.Tests.csproj`

File: src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/Microsoft.Azure.CognitiveServices.FormRecognizer.csproj [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  <!-- Please do not move/edit code below this line -->
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
+  <!-- Please do not move/edit code above this line -->
+
+  <ItemGroup>
+    <Folder Include="Properties\" />
+  </ItemGroup>
+
+  <PropertyGroup>
+    <TargetFrameworks>$(SdkTargetFx)</TargetFrameworks>
```
Comment (dsgouda) [inline src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/Microsoft.Azure.CognitiveServices.FormRecognizer.csproj]: Please refer [this](https://github.com/Azure/azure-sdk-for-net/blob/6e48acd9b5ad03fdf591f058417b434fdbdfd023/src/SDKs/CognitiveServices/dataPlane/Vision/ContentModerator/ContentModerator/Microsoft.Azure.CognitiveServices.ContentModerator.csproj) as an example and add the remaining attributes (Assembly version, package release notes, etc
Also, please set the assembly version to `0.8.0-preview`

File: src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/Properties/AssemblyInfo.cs [hunk 7]
```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Reflection;
+using System.Resources;
+
+[assembly: AssemblyTitle("Microsoft Cognitive Services Form Recognizer SDK")]
+[assembly: AssemblyDescription("Provides access to the Microsoft Cognitive Services Form Recognizer APIs.")]
+
+[assembly: AssemblyVersion("1.0.0.0")]
```
Comment (dsgouda) [inline src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/Properties/AssemblyInfo.cs]: Please set version to `0.8.0-preview`

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer.Tests/Microsoft.Azure.CognitiveServices.FormRecognizer.Tests.csproj (hunk 5 above)
Comment (ramparab) [reply src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer.Tests/Microsoft.Azure.CognitiveServices.FormRecognizer.Tests.csproj]: done.

File: src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/Properties/AssemblyInfo.cs (hunk 7 above)
Comment (ramparab) [reply src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/Properties/AssemblyInfo.cs]: I get following build error:
Properties\AssemblyInfo.cs(10,28,10,43): error CS7034: The specified version string does not conform to the required format - major[.minor[.build[.revision]]]

Comment (dsgouda) [reply src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/Properties/AssemblyInfo.cs]: In the `AssemblyInfo.cs` this should be set to `0.8.0.0`

File: src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/Microsoft.Azure.CognitiveServices.FormRecognizer.csproj (hunk 6 above)
Comment (ramparab) [reply src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/Microsoft.Azure.CognitiveServices.FormRecognizer.csproj]: done.

File: src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/Microsoft.Azure.CognitiveServices.FormRecognizer.csproj [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+	</PackageReleaseNotes>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Folder Include="Properties\" />
+  </ItemGroup>
+
+  <PropertyGroup>
+    <TargetFrameworks>$(SdkTargetFx)</TargetFrameworks>
+    <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
```
Comment (dsgouda) [inline src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/Microsoft.Azure.CognitiveServices.FormRecognizer.csproj]: Please remove lines 25-27

File: src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/Properties/AssemblyInfo.cs [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Reflection;
+using System.Resources;
+
+[assembly: AssemblyTitle("Microsoft Cognitive Services Form Recognizer SDK")]
+[assembly: AssemblyDescription("Provides access to the Microsoft Cognitive Services Form Recognizer APIs.")]
+
+[assembly: AssemblyVersion("0.8.0.0")]
+[assembly: AssemblyFileVersion("1.0.0.0")]
```
Comment (dsgouda) [inline src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/Properties/AssemblyInfo.cs]: `AssemblyFileVersion` must also be set to `0.8.0.0`

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks good for the most part.

**Comment by dsgouda** [comment]
@ramparab Please link the rest spec PR too

File: src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/Microsoft.Azure.CognitiveServices.FormRecognizer.csproj (hunk 8 above)
Comment (ramparab) [reply src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/Microsoft.Azure.CognitiveServices.FormRecognizer.csproj]: done.

File: src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/Properties/AssemblyInfo.cs (hunk 9 above)
Comment (ramparab) [reply src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/Properties/AssemblyInfo.cs]: done.

File: src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/Microsoft.Azure.CognitiveServices.FormRecognizer.csproj (hunk 6 above)
Comment (ramparab) [reply src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/Microsoft.Azure.CognitiveServices.FormRecognizer.csproj]: done.

File: src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/Properties/AssemblyInfo.cs (hunk 7 above)
Comment (ramparab) [reply src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/Properties/AssemblyInfo.cs]: done.

Comment (ramparab) [reply src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/Properties/AssemblyInfo.cs]: done.

File: src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/Microsoft.Azure.CognitiveServices.FormRecognizer.csproj (hunk 6 above)
Comment (ramparab) [reply src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/Microsoft.Azure.CognitiveServices.FormRecognizer.csproj]: done.

**Comment by ramparab** [comment]
FR REST Spec PR: 
https://github.com/Azure/azure-rest-api-specs/pull/5684

File: src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/generate.ps1 (hunk 4 above)
Comment (dsgouda) [reply src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/generate.ps1]: If the spec is merged, please regenerate the code using this script and commit all the changes generated. We expect a .txt and an SdkInfo_*.cs file to be generated.

Comment (ramparab) [reply src/SDKs/CognitiveServices/dataPlane/Vision/FormRecognizer/FormRecognizer/generate.ps1]: done.

**Review by dsgouda — APPROVED** [review]
LGTM will merge once CIs pass

**Pull request merged**